crates/bevy_ecs/src/system/mod.rs

```diff
@@ -276,6 +276,60 @@ mod tests {
         assert_eq!(*world.resource::<SystemRan>(), SystemRan::Yes);
     }
 
+    #[test]
+    fn get_many_is_ordered() {
+        use crate::system::Resource;
+        const ENTITIES_COUNT: usize = 1000;
+
+        #[derive(Resource)]
+        struct EntitiesArray(Vec<Entity>);
+
+        fn query_system(
+            mut ran: ResMut<SystemRan>,
+            entities_array: Res<EntitiesArray>,
+            q: Query<&W<usize>>,
+        ) {
+            let entities_array: [Entity; ENTITIES_COUNT] =
+                entities_array.0.clone().try_into().unwrap();
+
+            for (i, w) in (0..ENTITIES_COUNT).zip(q.get_many(entities_array).unwrap()) {
+                assert_eq!(i, w.0);
+            }
+
+            *ran = SystemRan::Yes;
+        }
+
+        fn query_system_mut(
+            mut ran: ResMut<SystemRan>,
+            entities_array: Res<EntitiesArray>,
+            q: Query<&mut W<usize>>,
+        ) {
+            let entities_array: [Entity; ENTITIES_COUNT] =
+                entities_array.0.clone().try_into().unwrap();
+
+            #[allow(unused_mut)]
+            for (i, mut w) in (0..ENTITIES_COUNT).zip(q.get_many(entities_array).unwrap()) {
```

Comment:
This doesn't seem right: shouldn't this be testing get_many_mut?

Reply:
You are right. My mistake.